GetStream/stream-chat-swift
# [CIS-1829] Call `channel/read` endpoint only when unread

### 🔗 Issue Links

CIS-1829

### 🎯 Goal

Fix regression introduced in https://github.com/GetStream/stream-chat-swift/pull/1978 and not spam `channel/read` endpoint if the channel is fully read

### 🛠 Implementation

Check the current user's channel read before calling the endpoint

### 🧪 Manual Testing Notes

1. Run `DemoApp`
2. Sign in as any user
3. Open any channel
4. Scroll it up & down multiple times, send new messages

Expected result:
No API rate limit errors for the `channel/read` endpoint reported

### ☑️ Contributor Checklist


### 🎁 Meme
<img width="1380" alt="Screenshot 2022-05-06 at 19 12 54" src="https://user-images.githubusercontent.com/12818985/167161758-509722ce-2e1c-47a3-9805-b6dc2e11af4e.png">
